source/index-management.rst

```diff
@@ -185,6 +212,48 @@ Standard API
     listIndexes(): Cursor;
   }
 
+  interface CreateIndexOptions {
+    /**
+     * Specifies how many data-bearing members of a replica set, including the primary, must
+     * complete the index builds successfully before the primary marks the indexes as ready.
+     *
+     * This option accepts the same values for the "w" field in a write concern plus "all",
+     * which indicates all voting data-bearing nodes.
+     *
+     * Drivers MUST NOT throw an error if this option is specified when creating an index on
```

Comment:
If we're making an effort to keep things consistent with the CRUD spec, can this be changed to use wording from there as well? For example, for the `hint` option on `Delete` helpers, the CRUD spec says:
```
This option is only supported by servers >= 4.4. Older servers >= 3.4 will report an error for using this option.
```
If you prefer the wording here instead, let's file a follow-on DRIVERS ticket to make CRUD consistent?

Reply:
So I included the language about not manually throwing an error in accordance with the [mantra](https://github.com/mongodb/specifications#where-possible-depend-on-server-to-return-errors), but maybe I should relax that to only apply to 3.4+? Ideally, we could give the mantras some teeth by actually enforcing them with "MUST", but I'm not sure if it would be frustrating for users to get different types of errors for the same thing on different server versions. Given this is somewhat of an obscure option, I don't think it'll matter in this case, but it may set a precedent that could be used on a more highly used option.

Alternatively, it could just be exactly consistent with the CRUD spec.

Reply:
What does a 3.2 server do?

Reply:
Doesn't look like we have 3.2 binaries for Ubuntu 18.04, so I can't verify locally, but I imagine it just silently ignores it.

Reply:
Tested on 3.2.22 and 3.4.24 locally. 3.2 does silently ignore the option. This is unfortunate, but I do think we should stay consistent with the CRUD spec here and say something like:

```
   * This option is only supported by servers >= 4.4. Older servers >= 3.4 will report an error for using this option.
   * For servers < 3.4, the driver MUST raise an error if the caller explicitly provides a value.
```

It will cause different types of errors depending on the operation, but I think that's better than specifying `commitQuorum` and having it silently do nothing.

Reply:
Since we're doing server version checks, I think we might want to manually through errors for the buggy servers (versions 4.2-4.2.5). Apparently, setting this option could [break things silently](https://jira.mongodb.org/browse/SERVER-47193?focusedCommentId=3019605&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3019605). The only problem with that is I think drivers only verify server version via maxWireVersion, so that would be we'd have to manually throw errors for all of 4.2. At that point, it might just be easiest to say that a driver should throw an error if server version is < 4.4...

Reply:
Given that there's an actual danger to users, I would say I'm now leaning towards requiring manual validation for this option all the time, and that we can rely on the server for validation of all other options.

Reply:
@patrickfreed let's not do any client side validation for this, the danger to users is relatively low and will be documented in the FCV documentation. Also, now that the server is patching the issue for 4.2, it will likely be ported to 4.4's FCV as well. I'm going to approve this now, let's move on to implementation.

Reply:
After offline discussion, we decided to require manual driver validation, since specifying the option could result in problems for users on 4.2.0-4.2.5. Updated the spec to reflect that.